src/cps/pc_compat/pc_base.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+// Copyright © 2024 Haowen Liu
+
+// This header is created to avoid writing non-trivial C++ code in pc.y.
+// This header contains types and declarations that are needed by both loader and parser.
+
+#pragma once
+
+#include <optional>
+#include <ostream>
+#include <string>
+#include <variant>
+#include <vector>
+
+namespace cps::pc_compat {
+    class PcLoader;
+
+    // The following types needs to be declared in the parser because
+    // PackageRequirement is the type of a non-terminal. bison needs
+    // to do a sizeof on this type and cannot do so with a forward
+    // declaration.
+    enum class VersionOperation { LT, LE, NE, EQ, GT, GE };
```

Comment:
```suggestion
    enum class VersionOperation { lt, le, ne, eq, gt, ge};
```

It's good practice to *always* prefix `YELLING_CASE` variables with a reasonably unique prefix. Ideally the project name since that's also self-documenting ownership. This prevents collisions with arbitrary `#define GE "General Electric"` or `#define EQ ==` other nonsense like that.

Options:

* Don't use `UPPER_CASE` variable names (my choice here)
* Use names like `CPS_CONFIG_LT`, etc.

Of course, do renames in an IDE with a refactoring feature instead of just applying this suggestion unless you want to have fun tracking down all the references a less efficient way.